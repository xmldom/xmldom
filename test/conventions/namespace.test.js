'use strict'

const { NAMESPACE } = require('../../lib/conventions')

describe('NAMESPACE', () => {
	Object.keys(NAMESPACE).forEach((key) => {
		if (key === 'isHTML') {
			describe('isHTML', () => {
				it('should return true for NAMESPACE.HTML', () => {
					expect(NAMESPACE.isHTML(NAMESPACE.HTML)).toBe(true)
				})
				it("should return true for 'http://www.w3.org/1999/xhtml'", () => {
					expect(NAMESPACE.isHTML('http://www.w3.org/1999/xhtml')).toBe(true)
				})
				it.each([undefined, null, 0, 1, false, true, '', NAMESPACE.XML])(
					"should return false for '%s'",
					(value) => {
						expect(NAMESPACE.isHTML(value)).toBe(false)
					}
				)
			})
		} else {
			const value = NAMESPACE[key]
			it(`should contain immutable ${key} with correct value`, () => {
				expect([key, value]).toMatchSnapshot()
				try {
					NAMESPACE[key] = 'boo'
				} catch {}
				expect(NAMESPACE[key]).toBe(value)
			})
		}
	})
})
