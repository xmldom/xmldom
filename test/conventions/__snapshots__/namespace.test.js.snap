// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAMESPACE should contain immutable HTML with correct value 1`] = `
Array [
  "HTML",
  "http://www.w3.org/1999/xhtml",
]
`;

exports[`NAMESPACE should contain immutable SVG with correct value 1`] = `
Array [
  "SVG",
  "http://www.w3.org/2000/svg",
]
`;

exports[`NAMESPACE should contain immutable XML with correct value 1`] = `
Array [
  "XML",
  "http://www.w3.org/XML/1998/namespace",
]
`;

exports[`NAMESPACE should contain immutable XMLNS with correct value 1`] = `
Array [
  "XMLNS",
  "http://www.w3.org/2000/xmlns/",
]
`;
